test/e2e/delete_test.go

```diff
@@ -4,47 +4,45 @@ import (
+	mapi "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
 	framework "github.com/operator-framework/operator-sdk/pkg/test"
 	"github.com/stretchr/testify/require"
-	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
 )
 
 func deletionTestSuite(t *testing.T) {
 	t.Run("Deletion", func(t *testing.T) { testWindowsNodeDeletion(t) })
-	t.Run("Status", func(t *testing.T) { testStatusWhenSuccessful(t) })
-	t.Run("ConfigMap validation", func(t *testing.T) { testConfigMapValidation(t) })
-	t.Run("Secrets validation", func(t *testing.T) { testValidateSecrets(t) })
 }
 
 // testWindowsNodeDeletion tests the Windows node deletion from the cluster.
 func testWindowsNodeDeletion(t *testing.T) {
 	testCtx, err := NewTestContext(t)
 	require.NoError(t, err)
 
-	// get WMCO custom resource
-	wmco := &operator.WindowsMachineConfig{}
-	// Get the WMCO resource called instance
-	err = framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: wmcCRName,
-		Namespace: testCtx.namespace}, wmco)
-	if err != nil && k8serrors.IsNotFound(err) {
-		// We did not find WMCO CR, let's recreate it. This is a possibility when the creation and deletion tests are
-		// run independently.
-		wmco, err = testCtx.createWMC(gc.numberOfNodes, gc.sshKeyPair)
-		require.NoError(t, err)
-	}
+	// The second item in the slice is the Windows MachineSet we're interested in scaling down.
+	require.Len(t, gc.machineSetNames, 2)
+	windowsMachineSet := &mapi.MachineSet{}
+	err = framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: gc.machineSetNames[1],
+		Namespace: "openshift-machine-api"}, windowsMachineSet)
 	// Reset the number of nodes to be deleted to 0
 	gc.numberOfNodes = 0
 	// Delete the Windows VM that got created.
-	wmco.Spec.Replicas = gc.numberOfNodes
-	if err := framework.Global.Client.Update(context.TODO(), wmco); err != nil {
-		t.Fatalf("error updating wcmo custom resource  %v", err)
+	windowsMachineSet.Spec.Replicas = &gc.numberOfNodes
+	if err := framework.Global.Client.Update(context.TODO(), windowsMachineSet); err != nil {
+		t.Fatalf("error updating windowsMachineSet custom resource  %v", err)
 	}
-	// As per testing, each windows VM is taking roughly 12 minutes to be shown up in the cluster, so to be on safe
-	// side, let's make it as 60 minutes.
-	err = testCtx.waitForWindowsNodes(gc.numberOfNodes, true)
+	// As per testing, each windows VM is taking roughly 12 minutes to be shown up in the cluster
+	err = testCtx.waitForWindowsNodes(gc.numberOfNodes, true, true)
 	if err != nil {
 		t.Fatalf("windows node deletion failed  with %v", err)
 	}
+	// Clean up the windows MachineSets created by the test suite
+	for _, machineSetName := range gc.machineSetNames {
+		machineSet := &mapi.MachineSet{}
+		err = framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: machineSetName,
+			Namespace: "openshift-machine-api"}, machineSet)
```

Comment:
Is the `Get` needed? You can populate `machineSet` with the `name` and `namespace` and that suffice for deletion.

Reply:
It is always better to get latest version of CR or else delete will fail